### ISSUE TITLE
Change Request: Update issue templates to add checkboxes for LLM usage

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#20355]
### ESLint version

HEAD

### What problem do you want to solve?

Lately we've seen an uptick in AI-generated issues and PRs.

These range from unhelpful, where the person clearly didn't even review what was generated:
https://github.com/eslint/markdown/issues/587
https://github.com/eslint/markdown/pull/590

To helpful:
https://github.com/eslint/csstree/issues/100

The trouble is that it's not always clear if the issue or PR is AI-generated, because if it is, we'll want to treat it more carefully. Also, we'll want to make sure it's not a waste of time because someone opened it without even reading what was generated.

### What do you think is the correct solution?

I think we should add fields to our issue and PR templates that ask people to declare if the issue/PR was generated by AI. Something like:

**AI Acknowledgement**

[ ] I used AI to generate this issue/PR
[ ] I have reviewed AI-generated the content before submitting

### Participation

- [ ] I am willing to submit a pull request for this change.

### Additional comments

I feel like we need to get ahead of this as I'm sure we'll be seeing a lot more in the future.

**Comment by fasttime** [comment]
Makes sense to me. I would like to hear the perspective of other team members before marking as accepted.

For reference, #20378 is a more recent example of an AI-generated issue.

**Comment by kirkwaiblinger** [comment]
I'm observing this issue out of interest, since we have similar concerns with AI-generated content at typescript-eslint. 

Would it make sense to have the AI disclosure be _invalid_ by default in the template? As in 

```
[ ] I did _not_ use AI to generate this issue/PR
[ ] (if above not checked) I have reviewed AI-generated content before submitting
```

(a valid response would involve at least one checkmark)

Just thinking to guard against AI generated content where the author didn't even bother to read the template and therefore would fail to disclose.

**Comment by lumirlumir** [comment]
I'm also +1 on adding a checkbox for LLM usage.

For example, the Renovate repository includes an `AI assistance disclosure` section like this:

https://github.com/renovatebot/renovate/blob/main/.github/pull_request_template.md?plain=1#L16-L28

```md
## AI assistance disclosure


<!-- We request this information to assist reviewers in identifying AI-generated errors and other issues specific to AI usage. While we typically permit the use of AI tools, we appreciate being notified when they are employed. -->


Did you use AI tools to create any part of this pull request?


Please select one option and, if yes, briefly describe how AI was used (e.g., code, tests, docs) and which tool(s) you used.


- [ ] No — I did not use AI for this contribution.
- [ ] Yes — minimal assistance (e.g., IDE autocomplete, small code completions, grammar fixes).
- [ ] Yes — substantive assistance (AI-generated non‑trivial portions of code, tests, or documentation).
- [ ] Yes — other (please describe):
```

So, I think it might be helpful to discuss the exact template we want to use here.

---

Also, I can work on this issue and apply the change across the repositories if it's accepted.

**Comment by DMartens** [comment]
I think the problem for the AI or low-effort issues / PRs stems primarily from first time contributers, so I do not think we should add checkboxes to the default templates.
In the contributing guide we could explicitly state add that any form of low-effort issue / PR (which includes pure AI output) is rejected.
Further we add an instruction what first time contributors have to include in the issue / PR to show that they read the contributing guide. For example `typescript-eslint` has a note to [add an emoji](https://typescript-eslint.io/contributing/pull-requests#asking-questions).
This allows us to quickly filter who actually read it (people committing pure AI PRs are unlikely to have read them).

**Comment by kirkwaiblinger** [comment]
> For example `typescript-eslint` has a note to [add an emoji](https://typescript-eslint.io/contributing/pull-requests#asking-questions). This allows us to quickly filter who actually read it (people committing pure AI PRs are unlikely to have read them).

Counterexample: https://github.com/typescript-eslint/typescript-eslint/pull/11832#issue-3708927080

**Comment by michaelgiraldo** [comment]
🍍

**Comment by lumirlumir** [comment]
Friendly ping @eslint/eslint-team. What steps do we need to take next here?

**Comment by fasttime** [comment]
**TSC Summary:** This issue proposes adding checkboxes to issue and PR templates to better indicate if and how a contribution was AI generated. There is currently no definitive consensus from the team about this change.

**TSC Question:** Should this proposal be accepted?

**Comment by nzakas** [comment]
@kirkwaiblinger 
> I'm observing this issue out of interest, since we have similar concerns with AI-generated content at typescript-eslint.
> 
> Would it make sense to have the AI disclosure be _invalid_ by default in the template? As in
> 
> ```
> [ ] I did _not_ use AI to generate this issue/PR
> [ ] (if above not checked) I have reviewed AI-generated content before submitting
> ```
> 
> (a valid response would involve at least one checkmark)
> 
> Just thinking to guard against AI generated content where the author didn't even bother to read the template and therefore would fail to disclose.

I like this. :+1:

**Comment by lumirlumir** [comment]
The TSC decided at its meeting on `UTC Thu 08-Jan-2026 21:00` to accept this and discuss further details more. So, I'm marking this as accepted.

---

I'll first update the `eslint` repository, then update the other repositories under `eslint` org once a decision is made about the template in https://github.com/eslint/eslint/pull/20431.

**Comment by lumirlumir** [comment]
Tracking:

- `.github`: https://github.com/eslint/.github/pull/18
- `code-explorer`: https://github.com/eslint/code-explorer/pull/321
- `css`: https://github.com/eslint/css/pull/382
- `eslint.org`: https://github.com/eslint/eslint.org/pull/943
- `eslint`: https://github.com/eslint/eslint/pull/20431
- `js`: https://github.com/eslint/js/pull/729
- `json`: https://github.com/eslint/json/pull/215
- `markdown`: https://github.com/eslint/markdown/pull/622
- `rewrite`: https://github.com/eslint/rewrite/pull/374

**Comment by lumirlumir** [comment]
All repositories with PR or issue templates have successfully adopted AI usage checkboxes, I'm closing this issue as complete.
